txn.go

```diff
@@ -186,18 +189,31 @@ type RunnerParams struct {
 	// Clock is an optional clock to use. If Clock is nil, clock.WallClock will
 	// be used.
 	Clock Clock
+
+	// ServerSideTransactions indicates that if SSTXNs are available, use them.
+	// Note that we will check if they are supported server side, and fall
+	// back to client-side transactions if they are not supported.
+	ServerSideTransactions bool
 }
 
 // NewRunner returns a Runner which runs transactions for the database specified in params.
 // Collection names used to manage the transactions and change log may also be specified in
 // params, but if not, default values will be used.
 func NewRunner(params RunnerParams) Runner {
+	sstxn := params.ServerSideTransactions
+	if sstxn {
+		sstxn = SupportsServerSideTransactions(params.Database)
+		if !sstxn {
+			logger.Infof("server-side transactions requested, but database does not support them")
```

Comment:
My instinct is to log it at warning level and also note that client-side transactions will be used.